docker-compose.yml

```diff
@@ -0,0 +1,22 @@
+version: "3.7"
+
+x-common-props: &common-props
+  build:
+    context: ./
+    target: base
+  working_dir: /base
+  volumes:
+    - ./:/base
+    - node_modules:/base/node_modules
+
+services:
+  npm:
+    <<: *common-props
+    entrypoint: npm
+  dev:
+    <<: *common-props
+    ports:
+      - "3000:3000"
+    command: npm run dev
+volumes:
+  node_modules:
```

Comment:
Do we need a docker-compose file right at the start? I am imagining we Helm uses it but I don't know for sure.

Reply:
This is also from the example I linked in the previous comment. I would say that we don't _need_ it but I thought it looked like a convenient setup that just meant devs don't have to pass in options to docker to map the volumes or expose the ports.

Reply:
This is interesting why the seperate npm container ?

Reply:
I thought I understood the reason for it but I don't believe I do after some experimentation. So I removed it in the last commit.